api/src/main/java/javax/enterprise/inject/spi/builder/ObserverMethodBuilder.java

```diff
@@ -0,0 +1,231 @@
+     * If the builder declares the {@link Default} qualifier, it's automatically removed.
+     *
+     * @param qualifiers to add to event
+     * @return self
+     */
+    ObserverMethodBuilder<T> addQualifiers(Annotation... qualifiers);
+
+    /**
+     *
+     * Add all the qualifiers to the Observed event
+     * If the builder declares the {@link Default} qualifier, it's automatically removed.
+     *
+     * @param qualifiers to add to event
+     * @return self
+     */
+    ObserverMethodBuilder<T> addQualifiers(Set<Annotation> qualifiers);
+
+    /**
+     * Replace all qualifiers on the Observed event.
+     * If the builder declares the {@link Default} qualifier, it's automatically removed.
+     *
+     * @param qualifiers to put on event
+     * @return self
+     */
+    ObserverMethodBuilder<T> qualifiers(Annotation... qualifiers);
+
+    /**
+     * Replace all qualifiers on the Observed event.
+     * If the builder declares the {@link Default} qualifier, it's automatically removed.
+     *
+     * @param qualifiers to put on event
+     * @return self
+     */
+    ObserverMethodBuilder<T> qualifiers(Set<Annotation> qualifiers);
+
+    /**
+     *
+     * Add a stereotype to the Observed parameter
+     *
+     * @param stereotype to add to event
+     * @return self
+     */
+    ObserverMethodBuilder<T> addStereotype(Class<? extends Annotation> stereotype);
+
+    /**
+     *
+     * Add all the stereotypes to the Observed parameter
+     *
+     * @param stereotypes to add to event
+     * @return self
+     */
+    ObserverMethodBuilder<T> addStereotypes(Set<Class<? extends Annotation>> stereotypes);
+
+    /**
+     *
+     * Replace all the stereotypes on the Observed parameter
+     *
+     * @param stereotypes to put on event
+     * @return self
+     */
+    ObserverMethodBuilder<T> stereotypes(Set<Class<? extends Annotation>> stereotypes);
+
+    /**
+     *
+     * Set the {@link Reception} mode for the observer to build
+     *
+     * @param reception reception type
+     * @return self
+     */
+    ObserverMethodBuilder<T> reception(Reception reception);
+
+    /**
+     *
+     * Set the {@link TransactionPhase} for the observer to build
+     *
+     * @param transactionPhase phase for the observer
+     * @return self
+     */
+    ObserverMethodBuilder<T> transactionPhase(TransactionPhase transactionPhase);
+
+
+    /**
+     *
+     * Set the priority for the observer to build
+     *
+     * @param priority priority of the observer
+     * @return self
+     */
+    ObserverMethodBuilder<T> priority(int priority);
+
+    /**
+     *
+     * Define consumer to call when event is notified
+     *
+     * @param consumer to call for the event notification
+     * @return self
+     */
+    ObserverMethodBuilder<T> notifyWith(Consumer<T> consumer);
+
+    /**
+     *
+     * Make the observer to build asynchronous.
+     * When no specified the observer is asynchronous.
```

Comment:
+1 Maybe I would change the wording to something similar:
_Make the observer method asynchronous. By default, the resulting observer method is synchronous._